drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_priv.h

```diff
@@ -53,6 +53,16 @@ struct lpspi_data {
 	size_t transfer_len;
 };
 
+
+/* Issuing a TCR is requirement on some version of LPSPI or last RX DMA transfer will
+ * never end. It also depends on flag SPI_HOLD_ON_CS, if it is set, we will try to
+ * keep PCS asserted during transanctions.
+ * On RT platform, main version is 01: TCR is needed.
+ * On MCXN platform, main version is 02: TCR is not needed.
+ */
+void spi_mcux_issue_TCR(const struct device *dev);
+
```

Comment:
why is this put on this file if it only seems to be used by the dma based driver ? Also, the comment is not clear about what this function actually *does*. The other functions in this file have comments that maybe are not super great but at least they describe what they do to some degree, this comment just describes a requirement, but it's not clear what "issue a TCR" means. I actually think this does not belong (at least yet) in the common header, if it's specific to the DMA driver. However, I noticed the HOLD_ON_CS flag is not being fixed for the cpu-based driver. Is that not intended for this PR?

Reply:
I will look into this flag for CPU based driver as part of https://github.com/zephyrproject-rtos/zephyr/pull/87789